robot_calibration/include/robot_calibration/optimize_parameters.h

```diff
@@ -0,0 +1,71 @@
+#ifndef ROBOT_CALIBRATION_OPTIMIZE_PARAMETERS_H
+#define ROBOT_CALIBRATION_OPTIMIZE_PARAMETERS_H
+
+
+/**
+* @brief      Verifies the required parameters have been set and, if not, set default values.
+*
+* @param[in]  node  node handle reference
+*
+* @return     bool indicating whether the parameters are all set.
+*/
+bool check_parameters(ros::NodeHandle& node);
+
+
+/**
+* @brief      Creates a string for the absoulte path for a file in the home directory.
+*
+* @param[in]  local_dir     name of the local directory.
+*
+* @return     string representing the absolute path to the file.
+*/
+std::string get_absolute_directory(std::string local_dir);
```

Comment:
```suggestion
std::string get_absolute_directory(const std::string& local_dir);
```

When you pass strings into a function it is a good practice to use const& or string_view ( if you have C++17 ).